e2e/nomostest/testutils/validate.go

```diff
@@ -40,3 +41,15 @@ func ValidateError(errs []v1beta1.ConfigSyncError, code, message string) error {
 	}
 	return errors.Errorf("error %s not present: %s", code, log.AsJSON(errs))
 }
+
+// WithFinalizer adds a finalizer to the object
+func WithFinalizer(obj client.Object, finalizer string) {
+	finalizers := obj.GetFinalizers()
+	finalizers = append(finalizers, finalizer)
+	obj.SetFinalizers(finalizers)
+}
+
+// WithoutFinalizers removes all finalizers from the object
+func WithoutFinalizers(obj client.Object) {
+	obj.SetFinalizers([]string{})
```

Comment:
Just nil it out, IMO.

Reply:
Or does that not work for SSA when you don't own the field?

Reply:
Since you're using SSA to append a finalizer, yes, nilling it out will remove it with SSA, even if it's just omitted.

Reply:
done